improver_tests/utilities/test_cube_checker.py

```diff
@@ -182,32 +187,37 @@ def test_coord_promotion_missing_scalar(self):
         """Test case in which a scalar coordinate has been lost from new_cube,
         meaning the cube undergoing checking ends up with different dimension
         coordinates to the progenitor cube. This raises an error."""
-        cube = set_up_cube()
-        new_cube = iris.util.squeeze(cube)
+        new_cube = iris.util.squeeze(self.cube)
         new_cube.remove_coord("realization")
         msg = "The number of dimension coordinates within the new cube"
         with self.assertRaisesRegex(CoordinateNotFoundError, msg):
-            check_cube_coordinates(cube, new_cube)
+            check_cube_coordinates(self.cube, new_cube)
 
 
 class Test_find_dimension_coordinate_mismatch(IrisTest):
 
     """Test if two cubes have the dimension coordinates."""
 
+    def setUp(self):
+        """Set up a cube."""
+        data = np.ones((2, 16, 16), dtype=np.float32)
+        data[:, 7, 7] = 0.0
```

Comment:
I don't think the data is actually ever used, so this data point setting is not needed.

Reply:
Removed.